### ISSUE TITLE
scala3.library.3: Invalid module name: '3' is not a Java identifier

### ISSUE THREAD

**Issue by makingthematrix** [lampepfl/dotty#12899]
## Compiler version

3.0.0

## Minimized code

in `pom.xml` add to dependencies:
```
<dependency>
            <groupId>org.scala-lang</groupId>
            <artifactId>scala3-library_3</artifactId>
            <version>3.0.0</version>
</dependency>
```
and then try to build the app with `mvn`. (please read on for detailed explanation).

## Output

```
[WARNING] There are 1 pathException(s). The related dependencies will be ignored.
[WARNING] Dependency: /home/makingthematrix/.m2/repository/org/scala-lang/scala3-library_3/3.0.0/scala3-library_3-3.0.0.jar
   - exception: Unable to derive module descriptor for /home/makingthematrix/.m2/repository/org/scala-lang/scala3-library_3/3.0.0/scala3-library_3-3.0.0.jar
   - cause: scala3.library.3: Invalid module name: '3' is not a Java identifier
```

## Expectation

There should be no warning about "'3' is not a Java identifier" and no later errors when the code tries to access the Scala library. 

## Details

This will happen in projects which use together Scala, Maven, and Java 9+. It's not a common combination, but  currently this is the way to write programs like [this one](https://github.com/makingthematrix/scalaonandroid/tree/main/HelloFXML3) - an Android app written in Scala 3 that uses JavaFX as GUI, and is built with GraalVM native image. Currently there is no way to do it with SBT. But Maven works pretty well. Except this one issue.

Since Java 9, the Scala way of adding a suffix with the Scala version at the end of artifact id of the library is invalid - hence the error above. An easy way to fix it is to add a line to `MANIFEST.MF` of the library's jar file:
```
Automatic-Module-Name: <library name> 
```

This can be done from SBT. For example in case of `scala3-library_3` it's enough to add something like this to `dotty/project/Build.scala`:
```
lazy val `scala3-library` = project.in(file("library")).
    asDottyLibrary(NonBootstrapped).
    settings(
      Compile / packageBin / packageOptions += Package.ManifestAttributes("Automatic-Module-Name" -> "scala3-library")
    )
  lazy val `scala3-library-bootstrapped`: Project = project.in(file("library")).
    asDottyLibrary(Bootstrapped).
    settings(
      Compile / packageBin / packageOptions += Package.ManifestAttributes("Automatic-Module-Name" -> "scala3-library")
    )
```
Please note that usually it's better to hardcode the value `scala3-library` instead of trying to derive it from the library name.

Actually, the same problem exists for all Scala 2 libraries. I wrote a Maven plugin that fixes this downstream - i.e. it unzips the downloaded jar file of the given Scala library, modifies `MANIFEST.MF`, and zips it again - but it would be better to have it fixed at the source. [Here's the link to the plugin page](https://github.com/makingthematrix/scala-suffix) where you can read more about this issue.

I'm planning to create a PR here in the Dotty project and when it's done I will link it to this bug report. This is the first time I do it, so sorry if I mess up something.

**Comment by smarter** [comment]
It seems to me that the most robust thing here would be to make a PR against sbt to have it automatically set Automatic-Module-Name based on the (organization + artifact name which should be unique) unless it's already set.

**Comment by makingthematrix** [comment]
@smarter : This is advised against, sometimes even in capital letters ;) https://users.scala-lang.org/t/scala-jdk-11-and-jpms/6102/38 . As far as I understand, the reason is that Automatic-Module-Name should be stable across library versions, and it's too easy to have a random modification if it's derived.

**Comment by smarter** [comment]
I think that warning is way too strict. It originally comes from https://github.com/akka/akka/commit/af411d2f8d2175b047c03354364b4c140d15d7eb which cites http://mail.openjdk.java.net/pipermail/jpms-spec-experts/2017-February/000582.html, I've read that email and I don't think any of the reason cited for not deriving the name automatically are very good.

> the reason is that Automatic-Module-Name should be stable across library versions, and it's too easy to have a random modification if it's derived.

If the name is only based on the organization and project name, then it won't randomly change, it's as stable as a project identifier can be. It would certainly be a lot more stable than not having Automatic-Module-Name and instead relying on the default behavior of deriving the name from the jar filename.

**Comment by makingthematrix** [comment]
@smarter : Okay, how about something like this?
```
  lazy val `scala3-library` = project.in(file("library")).
    asDottyLibrary(NonBootstrapped).
    settings(
      Compile / packageBin / packageOptions += 
        Package.ManifestAttributes("Automatic-Module-Name" -> s"$dottyOrganization-${name.value}")
    )
```

**Comment by smarter** [comment]
I think that's fine but what I'm really suggesting is you should make a PR against sbt to do this by default, then we can just upgrade sbt and won't have to worry about this :). (Though I don't mind doing it ourselves as a stop-gap if getting this in sbt ends up taking time).

**Comment by makingthematrix** [comment]
@smarter : that's an interesting idea, but I'm not sure if it's not too much and if I'm good enough with sbt to do it :)

Anyway, for now, here's my PR: https://github.com/lampepfl/dotty/pull/12903

I will think about making that PR to sbt as well.

**Comment by smarter** [comment]
I've opend https://github.com/sbt/sbt/issues/6561 to discuss this further.

**Comment by bishabosha** [comment]
fixed by #12903
